src/kernels/variables/pythonVariableRequester.node.ts

```diff
@@ -159,6 +159,9 @@ export class PythonVariablesRequester implements IKernelVariableRequester {
                   )
                 : [];
 
+            if (kernel.disposed || kernel.disposing) {
```

Comment:
On CI found plenty of errors where variable requestor would throw errors.
I suspect thats because the kernel is dead,
adding code here to check if kernel is dead/dying, and if thats the case then just exit with an empty list instead of falling over & polluting the logs with unnecessary errors.